Trev8939/create save map named user (#887)
Esri/arcgis-runtime-samples-android: java/create-and-save-map/README.md

> 

Reviewer: use case also has been updated to the following:

> Maps can be created programmatically in code and then serialized and saved as an ArcGIS portal item. In this case, the portal item is a web map which can be shared with others and opened in various applications and APIs throughout the platform, such as ArcGIS Pro, ArcGIS Online, the JavaScript API, Collector, and Explorer.

## How to use the sample

When you run the sample, you will be challenged for an ArcGIS Online login. Enter a user name and password for an ArcGIS Online named user account (such as your ArcGIS for Developers account). Then, choose the basemap and layers for your new map. To save the map, add a title, tags, description (optional), and a folder on your portal (you will need to create one in your portal's My Content section if you don't already have one). Click the Save button to save the map to the chosen folder.

## How it works

1. Set a `DefaultAuthenticationChallengeHandler` to the app's `AuthenticationManager`.

> 2. Create a new `Portal` and load it. Once it has loaded, an authentication challenge will be issued and you'll be prompted to enter your username and password.

Reviewer: another tweak from the design (changed since this was more user focussed than dev focussed) :D 

```suggestion
2. Create a new `Portal` and load it to invoke the authentication challenge.
```
